File: single-language-analysis/dataset-files/java/TestSmell128.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell128 {

    @Test
    public void testDirectionAndValueOfPinIsResetWhenClosed() {
        try (Pin pin = pins.pin(0, Out)) {
            pin.setValue(1);
        }

        gpioAdmin("export", 17, PullDown);
        try {
            assertEquals("0\n", contentOf("/sys/class/gpio/gpio17/value"));
            assertEquals("in\n", contentOf("/sys/class/gpio/gpio17/direction"));
        } finally {
            gpioAdmin("unexport", 17);
        }
    }
}

File: single-language-analysis/dataset-files/python/test_smell_128.py
# Original URL: https://github.com/romilly/quick2wire-python-api/blob/c5e21e9d804012efd9d214d18909034b4b898c96/quick2wire/test_gpio.py#L80-L89

class TestGPIO:

    def test_smell_128(self):
        with pins.pin(0, Out) as pin:
            pin.value = 1
        
        gpio_admin("export", 17, PullDown)
        try:
            assert content_of('/sys/class/gpio/gpio17/value') == '0\n'
            assert content_of('/sys/class/gpio/gpio17/direction') == 'in\n'
        finally:
            gpio_admin("unexport", 17)